### DISCUSSION TITLE
Http related metrics coming from Quarkus are different from Vertx

### DISCUSSION THREAD

**Discussion by ShubhamRwt** [quarkusio/quarkus#32789] [Q&A] [answered]
Hi, I have some question related to the quarkus micrometer metrics.

1. Why the  http related metrics coming from `quarkus` different then the http related related metrics coming through `vertx`  if Quarkus is based upon vertx?

2. We get this metric  `http_server_bytes_read_sum` from Quarkus but with vertx we get `http_server_bytes_read_total. Is `total` and `sum` the same thing?
 
3.  We get this metric `http_server_connections_seconds_active_count` from quarkus. What does `seconds` mean here.  Is it similar to the vertx metrics `http_server_requests_total`?

FYI - @ppatierno

[a comment by an automated account is not shown]

**Comment by ebullient** [comment, marked as answer]
Quarkus integrates micrometer much differently than vertx does. So while yes, Quarkus uses vertx as its underlying request engine, its integration with micrometer is very different and more direct.

The suffixes are provided by the Micrometer Prometheus Meter Registry.

- http_server_bytes_read_sum should be the same as http_server_bytes_read_total.
- http_server_connections_seconds_active_count is a long task timer. It shows how many seconds the longest running connection has been active for.
- There should _also_ be http_server_requests_total (and related) that present metrics for completed requests.

**Reply by ShubhamRwt** [reply]
```
# HELP strimzi_bridge_http_server_requests_seconds  
# TYPE strimzi_bridge_http_server_requests_seconds summary
strimzi_bridge_http_server_requests_seconds_count{method="POST",outcome="SUCCESS",code="200",path="/consumers/{groupid}",} 1.0
strimzi_bridge_http_server_requests_seconds_sum{method="POST",outcome="SUCCESS",code="200",path="/consumers/{groupid}",} 0.065377958
# HELP strimzi_bridge_http_server_requests_seconds_max  
# TYPE strimzi_bridge_http_server_requests_seconds_max gauge
strimzi_bridge_http_server_requests_seconds_max{method="POST",outcome="SUCCESS",code="200",path="/consumers/{groupid}",} 0.065377958
```

**Reply by ShubhamRwt** [reply]
These are all the `http_server_requests_seconds` related metrics

**Reply by ShubhamRwt** [reply]
There are no metrics as just `http_server_requests`. The `strimzi_bridge_` is a prefix we have added.

**Reply by ShubhamRwt** [reply]
@ebullient ^^^

**Reply by ebullient** [reply]
- http_server_requests_seconds_* -- Yes. that is the set for completed server requests.
- http_server_connections_seconds_active_* are long-task timers

The former measures completed requests (successful or not), the latter measures active/open connections.

**Reply by ShubhamRwt** [reply]
Thanks @ebullient
